### PR TITLE
Remove references to sun.* classes to be able to compile with --relea…

### PR DESCRIPTION
**Pull request by normanmaurer**
…se 8

Motivation:

The only reason why we need to cut our releases via java 8 is because we reference some sun.* classes. We can get rid of these by using MethodHandles / reflection

Modifications:

- Move sun.misc.Unsafe access to new class SunMiscUnsafeAccess and use MethodHandles to call it
- Refector OpenJdkSelfSignedCertGenerator to use MethodHandles / reflection

Result:

Be able to use --release 8 while building. Related to https://github.com/netty/netty/issues/14119

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6216,6 +6217,7 @@
           <fork>true</fork>
           <source>${maven.compiler.source}</source>
           <target>${maven.compiler.target}</target>
+          <release>${maven.compiler.release}</release>
```
Comment (chrisvest) [inline pom.xml:6220]: Are Java 8 builds fine with the presence of this setting?

File: handler/src/main/java/io/netty/handler/ssl/util/SelfSignedCertificate.java [hunk 2]
```diff
@@ -239,6 +239,7 @@ public SelfSignedCertificate(String fqdn, SecureRandom random, int bits, Date no
             // Try Bouncy Castle first as otherwise we will see an IllegalAccessError on more recent JDKs.
             paths = BouncyCastleSelfSignedCertGenerator.generate(
                     fqdn, keypair, random, notBefore, notAfter, algorithm);
+            throw new Exception();
```
Comment (chrisvest) [inline handler/src/main/java/io/netty/handler/ssl/util/SelfSignedCertificate.java:242]: Looks like testing left-over.

File: pom.xml (hunk 1 above)
Comment (normanmaurer) [reply pom.xml:6234]: Fixed

**Comment by normanmaurer** [comment]
Hmm seems like I need to fix some profile

**Comment by normanmaurer** [comment]
Alright... fixed by upgrading compiler plugin

**Comment by normanmaurer** [comment]
See also https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-release.html#usage-on-jdk-8

**Review by chrisvest — APPROVED** [review]

**Comment by chrisvest** [comment]
The Graal build broke:

```2024-06-21T21:36:16.6173275Z [INFO] --- exec:1.6.0:exec (verify-native-image) @ netty-testsuite-native-image ---
2024-06-21T21:36:16.6515812Z Exception in thread "main" com.oracle.svm.core.jdk.UnsupportedFeatureError: Invoke with MethodHandle argument could not be reduced to at most a single call or single field access. The method handle must be a compile time constant, e.g., be loaded from a `static final` field. Method that contains the method handle invocation: java.lang.invoke.LambdaForm$MH/645330631.invokeExact_MT(Object, Object, long, Object, Object)
2024-06-21T21:36:16.6519891Z 	at com.oracle.svm.core.util.VMError.unsupportedFeature(VMError.java:87)
2024-06-21T21:36:16.6521216Z 	at io.netty.util.internal.SunMiscUnsafeAccess.putObject(SunMiscUnsafeAccess.java:748)
2024-06-21T21:36:16.6522640Z 	at io.netty.util.internal.PlatformDependent0.putObject(PlatformDependent0.java:615)
2024-06-21T21:36:16.6525600Z 	at io.netty.util.internal.PlatformDependent.putObject(PlatformDependent.java:704)
2024-06-21T21:36:16.6526876Z 	at io.netty.channel.nio.NioIoHandler$3.run(NioIoHandler.java:194)
2024-06-21T21:36:16.6527996Z 	at java.security.AccessController.doPrivileged(AccessController.java:84)
2024-06-21T21:36:16.6529167Z 	at io.netty.channel.nio.NioIoHandler.openSelector(NioIoHandler.java:179)
2024-06-21T21:36:16.6530472Z 	at io.netty.channel.nio.NioIoHandler.<init>(NioIoHandler.java:120)
2024-06-21T21:36:16.6531619Z 	at io.netty.channel.nio.NioIoHandler.<init>(NioIoHandler.java:56)
2024-06-21T21:36:16.6532662Z 	at io.netty.channel.nio.NioIoHandler$4.newHandler(NioIoHandler.java:757)
2024-06-21T21:36:16.6533992Z 	at io.netty.channel.MultiThreadIoEventLoopGroup.newChild(MultiThreadIoEventLoopGroup.java:193)
2024-06-21T21:36:16.6535791Z 	at io.netty.channel.MultiThreadIoEventLoopGroup.newChild(MultiThreadIoEventLoopGroup.java:30)
2024-06-21T21:36:16.6537521Z 	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:84)
2024-06-21T21:36:16.6539239Z 	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:60)
2024-06-21T21:36:16.6540827Z 	at io.netty.channel.MultithreadEventLoopGroup.<init>(MultithreadEventLoopGroup.java:52)
2024-06-21T21:36:16.6542281Z 	at io.netty.channel.MultiThreadIoEventLoopGroup.<init>(MultiThreadIoEventLoopGroup.java:86)
2024-06-21T21:36:16.6543759Z 	at io.netty.channel.MultiThreadIoEventLoopGroup.<init>(MultiThreadIoEventLoopGroup.java:49)
2024-06-21T21:36:16.6545087Z 	at io.netty.testsuite.svm.HttpNativeServer.main(HttpNativeServer.java:42)
2024-06-21T21:36:16.7510849Z [ERROR] Command execution failed.
```

**Comment by normanmaurer** [comment]
@chrisvest I am confused about the error message as in fact the MethodHandle is `private static final ...`. Any idea ?

**Comment by normanmaurer** [comment]
@chrisvest PTAL again.. fixed the problem by upgrading graal. The previous used version didn't implement MethodHandle enough...

**Review by chrisvest — APPROVED** [review]

**Comment by chrisvest** [comment]
Looks good. Glad it was just a Graal bug fixed by upgrading.

**Pull request merged**